pkg/products/datasync/reconciler.go

```diff
@@ -111,6 +114,7 @@ func (r *Reconciler) reconcileTemplates(ctx context.Context, serverClient k8scli
 		if err != nil {
 			return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("failed to get file contents of %s: %w", templateFn, err)
 		}
+		defer fileData.Close()
```

Comment:
We could make sure that `getFileContentFromURL` closes it's request body and instead of a `Reader` just returns a string. Otherwise we always have to be careful to close the reader when using this function.